airflow/auth/managers/fab/security_manager/override.py

```diff
@@ -1432,23 +1430,14 @@ def get_oauth_user_info(self, provider, resp):
                 "last_name": data.get("family_name", ""),
                 "email": data.get("email", ""),
             }
-        # for Azure AD Tenant. Azure OAuth response contains
-        # JWT token which has user info.
-        # JWT token needs to be base64 decoded.
-        # https://docs.microsoft.com/en-us/azure/active-directory/develop/
-        # active-directory-protocols-oauth-code
         if provider == "azure":
-            log.debug("Azure response received : %s", resp)
-            id_token = resp["id_token"]
-            log.debug(str(id_token))
-            me = self._azure_jwt_token_parse(id_token)
-            log.debug("Parse JWT token : %s", me)
+            me = self._decode_and_validate_azure_jwt(resp["id_token"])
+            log.debug("User info from Azure: %s", me)
+            # https://learn.microsoft.com/en-us/azure/active-directory/develop/id-token-claims-reference#payload-claims
             return {
-                "name": me.get("name", ""),
-                "email": me["upn"],
+                "email": me["email"],
```

Comment:
Is this correct? After pulling latest main, this fails. Even if `email` is included in the scope claims it's not returned in the token, `upn` is included as before. (This could of course be an integration issue with the IDP on our end)

Either way, remember to include this in the release notes, because it may require updated scope in the webserver_config by the administrator.

Reply:
Well. This has been straight copied from FAB 4.3.9  - unless I made a mistake of course. We don't even analyse that we are just making sure that we have the same code as the original FAB Securiy Manager Has.

So if there is any problem with it - it should rather be raised to FAB not to us - our change is simply a `direct`  pass-through to FAB.

However this is a very valid point - I see that last week FAB released version 4.3.10 which contains this one: https://github.com/dpgaspar/Flask-AppBuilder/pull/2163   which solves precisely the problem you mention.

So ... time to upgradfe to 4.3.10 it seems.

Reply:
Merged in https://github.com/apache/airflow/pull/35991 -> will be released in 2.8.0